karapace/avro_compatibility.py

```diff
@@ -83,14 +97,21 @@ def merged_with(self, that: "SchemaCompatibilityResult") -> "SchemaCompatibility
             compatibility=compat, incompatibilities=incompatibilities, messages=messages, locations=locations
         )
 
+    def add_incompatibility(self, incompat_type: E, message: str, location: List[str]) -> None:
+        """Add an incompatibility, this will modify the object in-place."""
+        formatted_location = "/".join(location[1:] if len(location) > 1 else location)
+
+        self.compatibility = SchemaCompatibilityType.incompatible
+        self.incompatibilities.append(incompat_type)
+        self.messages.add(message)
+        self.locations.add(formatted_location)
+
     @staticmethod
     def compatible() -> "SchemaCompatibilityResult":
         return SchemaCompatibilityResult(SchemaCompatibilityType.compatible)
 
     @staticmethod
-    def incompatible(
-        incompat_type: SchemaIncompatibilityType, message: str, location: List[str]
-    ) -> "SchemaCompatibilityResult":
+    def incompatible(incompat_type: E, message: str, location: List[str]) -> "SchemaCompatibilityResult":
```

Comment:
Could from __future__ import annotations be useful here?

Reply:
I tend to use `__future__` imports only for forwards compatibility, which is not an issue anymore since python 2.7 reached its EOL. I don't see any problems with using it ... I also don't see any great advantages for the change here, so usually I just don't bother.